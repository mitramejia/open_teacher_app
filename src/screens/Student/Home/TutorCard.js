// =============================================================
//
// Name: TutorCard
// -> Description: 
//
// Author: mitramejia 
// Created at: 8/28/17
//
// =============================================================\\
import React from 'react';
import { Container, Content, Card, Header, Form, Item, Input, Toast, Button, Icon } from 'native-base';
import propTypes from 'prop-types';
import strings from './strings';
import style from './style';

class TutorCard extends React.Component {
  // render(){
  //   return (
  //     <Content>
  //     <Card>
  //       <CardItem header>
  //         <Text>NativeBase</Text>
  //       </CardItem>
  //       <CardItem>
  //         <Body>
  //         <Text>
  //
  //         </Text>
  //         </Body>
  //       </CardItem>
  //     </Card>
  //   </Content>
  // );
  // }
}