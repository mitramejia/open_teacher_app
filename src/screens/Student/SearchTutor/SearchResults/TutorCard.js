// =============================================================
//
// Name: TutorCard
// -> Description:
//
// Author: mitramejia
// Created at: 8/28/17
//
// =============================================================\\
import React from 'react';
import {
  Container,
  Content,
  Card,
  Header,
  Form,
  Item,
  Input,
  Toast,
  Button,
  Icon,
  Body,
  Text,
  ListItem,
  Thumbnail,
  Right,
} from 'native-base';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import strings from '../Form/strings';
// import style from '../Form/style';

class TutorCard extends React.Component {
  render() {
    const { user, job, rate, rating } = this.props.tutor;
    return (
      <Content>
        <ListItem avatar>
          <Body>
            <Text>
              {user.firstName} {user.lastName}
            </Text>
            <Text>
              rate: {rate}
            </Text>
            <Text>
              rating: {rating}
            </Text>
            <Text>
              job: {job}
            </Text>
          </Body>
        </ListItem>
      </Content>
    );
  }
}

TutorCard.PropTypes = {
  tutors: PropTypes.object.isRequired,
};

export default TutorCard;
