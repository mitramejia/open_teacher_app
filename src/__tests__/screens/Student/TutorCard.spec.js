import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Factory from '../../utils/factories';
import TutorCard from '../../../screens/Student/SearchTutor/SearchResults/TutorCard';

it('renders a Search Tutor Screen', () => {
  let tutor = Factory.build('tutor');
  const tree = renderer.create(<TutorCard tutor={tutor} />).toJSON();
  expect(tree).toMatchSnapshot();
});
