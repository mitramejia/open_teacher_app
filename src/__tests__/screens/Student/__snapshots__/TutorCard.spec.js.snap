// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Search Tutor Screen 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  virtual={undefined}
>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        avatar={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderBottomWidth": null,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 15,
            "paddingBottom": null,
            "paddingRight": null,
            "paddingTop": null,
          }
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": null,
              "alignSelf": null,
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flex": 1,
              "marginLeft": 15,
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
                "marginLeft": null,
                "marginRight": 10,
              }
            }
            uppercase={false}
            virtual={undefined}
          >
            Test
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            note={true}
            style={
              Object {
                "color": "#808080",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "200",
                "marginLeft": null,
                "marginRight": 10,
              }
            }
            uppercase={false}
            virtual={undefined}
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
